packages/ui-components/src/Inputs/TokenInput/TokenInput.tsx

```diff
@@ -117,36 +117,29 @@ const ChainName = styled.span`
 `;
 export const TokenInput: React.FC<TokenInputProps> = ({ currencies, onChange, value, wrapperStyles }) => {
   const selectItems = useMemo(() => {
-    return currencies.map((currency) => {
-      const view = currency.view;
+    const selectableItems = currencies.map((currency) => {
       return {
-        ...view,
+        ...currency.view,
         self: currency,
       };
     });
+
+    return selectableItems;
   }, [currencies]);
 
+  const selected = value ? { ...value.view, self: value } : null;
   useEffect(() => {
-    if (!value) {
-      onChange(currencies[0]);
+    if (value) {
+      onChange(value);
+      return;
     }
-  }, [value, currencies, onChange]);
+  }, [currencies, onChange, value]);
```

Comment:
This will create an infinite loop

  const selected = value ? { ...value.view, self: value } : null;
  useEffect(() => {
    if (value) {
      onChange(value);
      return;
    }
  }, [currencies, onChange, value]);